packages/iris-grid/src/mousehandlers/IrisGridPartitionedTableMouseHandler.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint class-methods-use-this: "off" */
+import {
+  type Grid,
+  GridMouseHandler,
+  GridPoint,
+  EventHandlerResult,
+} from '@deephaven/grid';
+import type IrisGrid from '../IrisGrid';
+import { isPartitionedGridModel } from '../PartitionedGridModel';
+
+/**
+ * Handles sending data selected via double click
+ */
+class IrisGridPartitionedTableMouseHandler extends GridMouseHandler {
+  constructor(irisGrid: IrisGrid) {
+    super(880);
+
+    this.irisGrid = irisGrid;
+  }
+
+  irisGrid: IrisGrid;
+
+  onDoubleClick(gridPoint: GridPoint, grid: Grid): EventHandlerResult {
+    const { column, row } = gridPoint;
+    if (
+      row == null ||
+      column == null ||
+      !isPartitionedGridModel(this.irisGrid.props.model)
+    ) {
+      return false;
+    }
+
```

Comment:
We also only want this to trigger when the `partitionConfig` is in `keys` mode; we don't want double-clicking on another table to trigger this behaviour. Get the `partitionConfig` from `irisGrid.state` and check it first, if it's not in `keys` mode then just `return false`.